File: common/src/main/java/com/pluxity/feature/dto/FeatureCreateRequest.java
package com.pluxity.feature.dto;

import com.pluxity.feature.entity.Spatial;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record FeatureCreateRequest(
        @NotBlank String id, // UUID
        Spatial position,
        Spatial rotation,
        Spatial scale,
        @NotNull Long assetId,
        @NotNull Long facilityId,
        @NotNull String floorId) {}

File: common/src/main/java/com/pluxity/feature/dto/FeatureCreateRequest.kt
package com.pluxity.feature.dto

import com.pluxity.feature.entity.Spatial
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class FeatureCreateRequest(
    // UUID
    @field:NotBlank
    val id: String,
    val position: Spatial?,
    val rotation: Spatial?,
    val scale: Spatial?,
    @field:NotNull
    var assetId: Long,
    @field:NotNull
    var facilityId: Long,
    val floorId: String?,
)
